Update latencymeter.h

KimIV opened the pull request:
Я проверил. Если убрать сравнение minTime с нулём, то LOW перестаёт скакать.

KimIV committed:
Update latencymeter.h

Я проверил. Если убрать сравнение minTime с нулём, то LOW перестаёт скакать.

KimIV committed:
Update latencymeter.h

Мне удалось убрать 2 в нулевой задержке, заменив millis() на micros() / 1000. Теперь в нулевой задержке фигурируют только 0 и 1. Думаю, с этим уже можно смириться...

KimIV committed:
Исправление нулевой задержки

Немного изменил расчёт valueTime, а также в переменной _timer теперь хранится micros(). Нулевая задержка теперь действительно нулевая!

githubniko commented:
> Я проверил. Если убрать сравнение minTime с нулём, то LOW перестаёт скакать.

Если так сделать, то Low всегда будет 0. Предлагаю при объявлении переменной minTime указать максимальное значение, тогда при первом измерении, minTime  всегда будет больше и примет первое значение. Либо что-то еще придумать.

В остальном отлично.

KimIV commented:
> Если так сделать, то Low всегда будет 0.

Да Вы правы! Я вспомнил, что решал уже подобные задачи двумя способами:
1. Если не используется область отрицательных значений, то инициализация переменной, хранящей LOW, значением -1. И далее сравнение minTime либо с текущим значением. либо с -1. То есть как у Вас было, только вместо нуля -1.
2. Именно как Вы предложили. Инициализация LOW максимально возможным значением для данного типа переменной.

Второй вариант мне представляется более универсальным и надёжным.

githubniko commented:
2ой вариант. Пришлите повторно запрос на слияние с изменениями. Спасибо.

KimIV commented:
Егор, я хотел уточнить для собственного понимания. Просто всегда имел дела только с форками, а своих мастер-проектов пока не вёл. Вот я отправил Вам запрос на слияние (Pull Request). В его составе сначала был один commit. Потом я добавил ещё один. И потом добавил третий. Если Вы будете сливать мои предложения, то у Вас есть выбор, какой именно коммит слить или можете только полностью запрос (Pull Request)? По другому спрошу. Вы можете в запросе выбирать, какой коммит слить со своим кодом, а какой игнорировать?

KimIV commented:
> 2ой вариант. Пришлите повторно запрос на слияние с изменениями. Спасибо.

Это хоббийная тема, ей только дома могу заниматься. Поэтому вечером после работы. Ну и после проверки естественно. Это будет часа через три-четыре...

githubniko commented:
> Егор, я хотел уточнить для собственного понимания. Просто всегда имел дела только с форками, а своих мастер-проектов пока не вёл. Вот я отправил Вам запрос на слияние (Pull Request). В его составе сначала был один commit. Потом я добавил ещё один. И потом добавил третий. Если Вы будете сливать мои предложения, то у Вас есть выбор, какой именно коммит слить или можете только полностью запрос (Pull Request)? По другому спрошу. Вы можете в запросе выбирать, какой коммит слить со своим кодом, а какой игнорировать?

Необходимо доработать или разбить на части. Я принимаю весь pull request.

> Это хоббийная тема, ей только дома могу заниматься. Поэтому вечером после работы. Ну и после проверки естественно. Это будет часа через три-четыре...

Хорошо. Не к спеху.

KimIV committed:
Исправление постоянного нуля в LOW

Переменную minTime инициировал максимальным для её типа значением 32767. И сделал это после вызова метода onUpdate(). Если это сделать до вызова метода onUpdate() и короткими нажатиями кнопкой настроить отображение LOW, то на короткое время мелькает значение 2767 (тройка не входит по разрядности). Поэтому я принял решение до вызова метода onUpdate() оставить minTime = 0, а после вызова вставить minTime = 32767

KimIV committed:
Исправил опечатки в комментах

KimIV commented:
Егор, готово!
